WordPress/Classes/ViewRelated/Blog/TimeZoneSelectorViewController.swift

```diff
@@ -0,0 +1,112 @@
+import UIKit
+
+class TimeZoneSelectorViewController: UITableViewController, ImmuTablePresenter {
+
+    /// users current timezone passed by SiteSettingsVC, if empty means manual offset is to be used
+    @objc var usersCurrentTimeZone: String?
+    /// users manual offset passed by SiteSettingsVC
+    @objc var usersManualOffset: NSNumber?
+
+    /// timezoneString will be set to a non empty string
+    /// if user selects anything other than Manual Offset section
+    /// manualOffset will be set if users selects anything from Manual Offset section
+    @objc var onChange: ((_ timezoneString: String, _ manualOffset: NSNumber?) -> Void)?
+
+    /// ImmuTableViewHandler, takes over the datasource, delegate from this VC
+    private lazy var handler: ImmuTableViewHandler = {
+        return ImmuTableViewHandler(takeOver: self)
+    }()
+
+    /// common action passed to TimezoneListRow cells, executed on select of a cell
+    private lazy var action: ImmuTableAction = { [weak self] (aRow) -> Void in
```

Comment:
You can make this a little more concise – `{ [weak self] row in`

Reply:
While the parentheses around `row` aren't required for closures (and I used to love that style), I think most of the instances I've seen use them, and Xcode autocompletes closures with them. 👍 on dropping `-> Void` though

Reply:
Also, the View Controller should not be the one implementing the action. The only reason `timezoneValue` needs to be stored in the row in this implementation is so that it can be matched here. If the action is moved closer to the row instantiation (in `TimezoneSelectorViewModel.tableViewModel()`), the timezone value is available in the context and can be captured by the closure. Or you could have an action constructor that takes the value as a parameter (e.g. `LanguageSelectorViewController.action(language:)`).